### DISCUSSION TITLE
Add an ability to index a single Assets folder (and its subfolders)

### DISCUSSION THREAD

**Discussion by andris-sevcenko** [craftcms/cms#8468] [Ideas]
It would be nice if it was possible to re-index just a single folder instead of having to always re-index the whole Volume.

**Comment by benface** [comment]
YES! Please! Also related to #2788

**Comment by jesuismaxime** [comment]
Any plan to add this to a further release ? We could use this feature (within the CLI) for a project that hold 125K assets dispatched in 10 subfolders on the same asset volume (AWS).

**Comment by brandonkelly** [comment]
We should be able to get it in soon.

**Comment by jesuismaxime** [comment]
> We should be able to get it in soon.

Looks like it's already possible. That works: `./craft index-assets/one {volume handle}/{subfolder}/{deeper subfolder}/`

Just some missing instructions in the CLI!

Thanks to @alexgilbertDG

**Comment by brandonkelly** [comment]
@jesuismaxime Amazing! We’ve improved the `index-assets/one` command for the next release. Guess we can close this now.

**Comment by jesuismaxime** [comment]
Is your improvement fix an issue when using `--delete-missing-assets=1` with `./craft index-assets/one {volume}/{subfolder}/{subfolder}` ? It deletes all the others volume's assets. I think that the command search for all assets on in this particular subfolder, since they "arent there".. they got deleted. I loose 15K+ assets yesterday.
